src/image/hfe.c

```diff
@@ -121,14 +124,20 @@ static bool_t hfe_open(struct image *im)
 static void hfe_seek_track(struct image *im, uint16_t track)
 {
     struct track_header thdr;
+    uint16_t old_len;
 
     F_lseek(&im->fp, im->hfe.tlut_base*512 + (track/2)*4);
     F_read(&im->fp, &thdr, sizeof(thdr), NULL);
 
     im->hfe.trk_off = le16toh(thdr.offset);
+    old_len = im->hfe.trk_len;
```

Comment:
I guess this is 0 on the first seek after mounting the image, so this should just work and cause stk_per_rev to be estimated on first seek?